open-craft/xblock-chat
# Allow user to respond in quick succession without transfering control to bot

This patch modifies the code to allow authors to define multiple user responses in succession without having to include bot messages in between.
The `messages` attribute is now allowed to be an empty list.

The patch also increases the maximum number of allowed responses from 3 to 7.

**Testing instructions**:

1. Add a new Chat block in the Studio and set `messages` to an empty list (`messages: []`) for some of the steps.
2. Verify that a problem defined in this way works correctly - when encountering steps with empty `messages` attribute, user response buttons should be rendered immediately.

**Note**:

This xblock stores state into `localStorage` in addition to the server, so when you're testing these changes, you will find that the "Delete learner's state" button from "Staff Debug" tools doesn't reset the problem. To completely reset the problem, you should invoke `localStorage.clear()` in the JS console in addition to invoking "Delete learner's state" staff debug tool.

**Reviewers**:

- [x] @replaceafill
